Refactor OCI CLI Auth Script for Modularity, Maintainability, and UX Improvements

rozdolsky33 opened the pull request:
# 🧹 Chore Pull Request

## Description  
This PR focuses on general cleanup, log standardization, and code quality improvements to support maintainability, clarity, and consistency between CLI tools and background processes.

---

## Key Improvements

### Modular Design  
- Introduced reusable helper functions:
  - `find_oci_auth_refresher_process`
  - `start_oci_auth_refresher`
  - `log_message`
- Centralized logic for detecting and managing background refresher processes (previously duplicated in multiple locations).

---

### Log Standardization  
- Implemented a consistent `log_message` function for all log output related to session management and process lifecycle.
- All messages written to `${HOME}/Library/Logs/oci-auth-refresher_<profile>.log` with timestamps.
- Ensures parity between `oshell.sh` and `oci_auth_refresher.sh`, enabling easier debugging and support.

**Purpose:**  
Improves observability of authentication lifecycle, process orchestration, and failure conditions across scripts.

---

### Improved Error Handling and Validation  
- Added validation for missing `OCI_CLI_PROFILE` during logout and status checks.  
- Validates existence of profile directories before setting or switching.  
- Uses consistent `return 0` / `return 1` codes across functions for better integration in pipelines.  

---

### Clearer Output and UX Enhancements  
- Improved terminal output formatting with color-coded messages.  
- Added usage guidance for commands like `ociset` and `oci_authenticate`.  
- Shows clearer status of background processes and session validation.

---

### Refresher Log File Handling  
- `REFRESHER_LOG_FILE` now dynamically updates based on active profile using `oci_set_profile`.  
- Previously hardcoded only once at the top of the script, which risked incorrect log file paths after profile switches.

---

### Cleaner Environment Variable Management  
- `oci_env_clear` now **preserves** `OCI_CLI_PROFILE` while unsetting other session-related vars.  
- Prevents users from inadvertently invalidating their session while switching context.  

---

### Logical Grouping of Code  
- Related functionality (e.g. env handling, auth, logout, status) grouped clearly.  
- Improved readability and navigation.

---

## ⚠️ Minor Behavioral Differences

| Feature             | Original Behavior                | Refactored Behavior                                |
|---------------------|----------------------------------|----------------------------------------------------|
| `oci_env_clear`     | Clears all variables             | Preserves `OCI_CLI_PROFILE`                        |
| `ociset`            | Basic export, no checks          | Adds profile existence check and usage warning     |
| `oci_auth_status`   | No unset profile check           | Warns if unset; outputs clearer session/process info|

---

## Summary  
-  Introduced static checks in CI for linting and consistency  
- Refactored and modularized codebase  
- Improved developer experience and process transparency  
- Log output now standardized for all background/CLI scripts  

---

## Checklist  
- [ ] No impact on functionality  
- [ ] All tests passing  
- [x] Static checks run locally and in CI *(currently failing — needs follow-up)*  
- [x] No docs updates needed

rozdolsky33 committed:
Add CI and local static checks to ensure code quality and maintainability.

rozdolsky33 committed:
Add CI badge

rozdolsky33 committed:
fix refresher command

rozdolsky33 committed:
Add troubleshooting and setup fix section to README for oci_auth_refresher.sh issues

rozdolsky33 committed:
Add table of contents for improved README navigation

dnlloyd commented on .github/workflows/ci.yml:
```diff
@@ -0,0 +1,50 @@
+        run: |
+          for script in $(find . -name '*.zsh' -o -name '*.sh'); do
+            echo "Testing script execution: $script"
+            zsh "$script" || exit 1
```
This will always fail for `oci_auth_refresher.sh` as it requires an argument. 

We could set a precedence that all scripts in this project require no arguments. I.e. Set sane defaults for all scripts that previously required input

E.g. Set `profile_name` to "DEFAULT" for `oci_auth_refresher.sh`

dnlloyd commented on README.md:
```diff
@@ -219,5 +311,151 @@ State: RUNNING
-
 </details>
+
+## Shell Integration
```
```suggestion
## Shell Prompt Integration
```

rozdolsky33 replied:
not sure where that came from. here is my current code. line 316
![Screenshot 2025-07-05 at 3 20 17 PM](https://github.com/user-attachments/assets/0bf2e275-4c01-481a-9d72-9d12c5015248)

dnlloyd commented on README.md:
```diff
@@ -219,5 +311,151 @@ State: RUNNING
+- Current OCI profile
+- Tenancy (if set)
+- Compartment (if set)
+
```
We should probably add a warning that this is a "beta" feature and may cause issues with your existing prompt and ZSH initialization.

dnlloyd commented on README.md:
```diff
@@ -219,5 +311,151 @@ State: RUNNING
+
+---
+
+## How It Works
```
```suggestion
## oshell's Authentication Refresher
```

rozdolsky33 replied:
![Screenshot 2025-07-05 at 3 24 32 PM](https://github.com/user-attachments/assets/46a775e9-96cd-4f28-b305-0fcb039dfd15)

dnlloyd commented on oci_auth_refresher.sh:
```diff
@@ -1,79 +1,123 @@
   fi
 }
 
-function refresh_session () {
```
NIT: Inconsistent space between function name and parentheses. I prefer no space 😁 but either way we should be consistent. 
Prob best to be POSIX compliant so: `refresh_session()` without the "function". But we can make that change later.

rozdolsky33 committed:
- Default to "DEFAULT" profile if none is provided in `oci_auth_refresher.sh`.
- Update `README.md` to reflect changes and clarify usage with/without profile name.

rozdolsky33 committed:
Clarify oshell installation path instructions in README and example `.zshrc`.
